projects/igniteui-angular/src/lib/combo/combo.common.ts

```diff
@@ -1154,6 +1154,9 @@ export abstract class IgxComboBaseDirective extends DisplayDensityBase implement
 
     protected findMatch = (element: any): boolean => {
         const value = this.displayKey ? element[this.displayKey] : element;
+        if (!value) {
```

Comment:
If value is `0`, this will be evaluated to `true`. Should scope the check only for `undefined` and `null`.

To see what issue causes this, go to combo.sample.html and only add this cmbo on the page:

``html`
                        <igx-combo #combo1 class="input-container" [itemsMaxHeight]="130" [data]="genres"
                            [valueKey]="'type'" [displayKey]="'type'" type="line" formControlName="genres"
                            searchPlaceholder="Search...">
                            <label igxLabel>States</label>
                        </igx-combo>
```
Modify the `genres` data collection to have a "0":

```typescript
        this.genres = [
            { type: '0' , movies: ['The Matrix', 'Kill Bill: Vol.1', 'The Dark Knight Rises']},
```

Run the sample and search in the combo for `0`